File: 3577-countTheNumberOfComputerUnlockingPermutations/countPermutations.hs
modV = 10 ^ 9 + 7

countPermutations :: [Int] -> Int
countPermutations (min : xs) =
  if smallest min xs
    then permutations (length xs)
    else 0
  where
    smallest m = all (< m)

    permutations 0 = 1
    permutations n = n * permutations (n - 1) `mod` modV

File: 3577-countTheNumberOfComputerUnlockingPermutations/countPermutations.rs
struct Solution;

impl Solution {
    fn permutations(n: usize) -> i32 {
        let mut result: i64 = 1;
        for i in 1..=n {
            result = (result * i as i64) % 1000000007;
        }
        result as i32
    }
    pub fn count_permutations(complexity: Vec<i32>) -> i32 {
        let min = complexity[0];
        if complexity.iter().skip(1).any(|&x| x <= min) {
            return 0;
        }
        Self::permutations(complexity.len() - 1)
    }
}
